lib/puma/dsl.rb

```diff
@@ -865,5 +865,22 @@ def nakayoshi_fork(enabled=true)
     def max_fast_inline(num_of_requests)
       @options[:max_fast_inline] = Float(num_of_requests)
     end
+
+    # Specify the backend for the IO selector.
+    #
+    # Provided values will be passed directly to +NIO::Selector.new+, with the
+    # exception of +:auto+ which will let nio4r choose the backend.
+    #
+    # Check the documentation of +NIO::Selector.backends+ for the list of valid
+    # options. The available options on your system will depend on the operating
+    # system.
+    #
+    # The default is +:auto+.
+    #
+    # @see https://github.com/socketry/nio4r/blob/master/lib/nio/selector.rb
```

Comment:
Should we also write something about the environment variable [`NIO4R_PURE`](https://github.com/socketry/nio4r/blob/v2.5.4/lib/nio.rb) here?

This was confusing to me before I learned about it:

```ruby
$ bundle e pry -rnio
[1] pry(main)> NIO::VERSION
=> "2.5.4"
[2] pry(main)> NIO::Selector.new(:ruby)
ArgumentError: unsupported backend: :ruby
from (pry):1:in `initialize'
```

```ruby
$ bundle e pry -rnio
[1] pry(main)> RUBY_PLATFORM
=> "x86_64-darwin18"
[2] pry(main)> NIO::VERSION
=> "2.5.4"
[3] pry(main)> [nil, :ruby, :java, :epoll, :poll, :kqueue, :select, :port, :unknown].map { |backend| NIO::Selector.new(backend).backend rescue nil }
=> [:kqueue, nil, nil, nil, :poll, :kqueue, :select, nil, nil]
```

```ruby
$ NIO4R_PURE=true bundle e pry -rnio
[1] pry(main)> RUBY_PLATFORM
=> "x86_64-darwin18"
[2] pry(main)> NIO::VERSION
=> "2.5.4"
[3] pry(main)> [nil, :ruby, :java, :epoll, :poll, :kqueue, :select, :port, :unknown].map { |backend| NIO::Selector.new(backend).backend rescue nil }
=> [nil, :ruby, nil, nil, nil, nil, nil, nil, nil]
```

Reply:
I agree. I might actually try to change that in nio4r. I'll document it.